velociraptor/observations/objects.py

```diff
@@ -569,16 +576,21 @@ def plot_on_axes(self, axes: Axes, errorbar_kwargs: Union[dict, None] = None):
         elif self.plot_as == "line":
             kwargs["zorder"] = line_zorder
 
-        # Make both the data name and redshift appear in the legend
-        data_label = f"{self.citation} ($z={self.redshift:.1f}$)"
+        # Check whether the data is available at several redshifts
+        if line_idx >= 0:
+            observational_data_y = self.y[line_idx, :]
+            data_label = f"{self.citation} ($z={self.redshift[line_idx]:.1f}$)"
+        else:
+            observational_data_y = self.y
+            data_label = f"{self.citation} ($z={self.redshift:.1f}$)"
 
         axes.errorbar(
             self.x,
```

Comment:
x may not be the same for all redshifts

Reply:
Well, yes you are right.

I will generalise it then to include the case wherein x are different for different z.

Reply:
Note the comment below about x and y are different _lengths_ for different z. I think we need to be much more careful about this before forging ahead with an implementation as we'll need to change quite a lot of files.